WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/Downloadable Files/ProductDownloadSettingsViewModelTests.swift

```diff
@@ -0,0 +1,259 @@
+import XCTest
+@testable import WooCommerce
+@testable import Yosemite
+
+final class ProductDownloadSettingsViewModelTests: XCTestCase {
+
+    // MARK: - Initialization
+
+    func test_readonly_values_are_as_expected_after_initializing_download_settings() {
+        // Arrange
+        let product = MockProduct().product(downloadable: true)
+        let model = EditableProductModel(product: product)
+
+        // Act
+        let viewModel = ProductDownloadSettingsViewModel(product: model)
+
+        // Assert
+        XCTAssertEqual(viewModel.downloadLimit, 1)
+        XCTAssertEqual(viewModel.downloadExpiry, 1)
+    }
+
+    func test_section_and_row_values_are_as_expected_after_initializing_download_settings() {
+        // Arrange
+        let product = MockProduct().product(downloadable: true)
+        let model = EditableProductModel(product: product)
+        let viewModel = ProductDownloadSettingsViewModel(product: model)
+
+        // Act
+        let expectedSections: [ProductDownloadSettingsViewController.Section] = [
+            .init(footer: ProductDownloadSettingsViewModel.Strings.downloadLimitFooter, rows: [ProductDownloadSettingsViewController.Row.limit]),
+            .init(footer: ProductDownloadSettingsViewModel.Strings.downloadExpiryFooter, rows: [ProductDownloadSettingsViewController.Row.expiry])
+        ]
+
+        // Assert
+        XCTAssertEqual(viewModel.sections, expectedSections)
+    }
+
+    // MARK: - `handleDownloadLimitChange`
+
+    func test_handling_empty_downloadLimit_updates_with_default_value() {
+        // Arrange
+        let product = MockProduct().product(downloadable: true)
+        let model = EditableProductModel(product: product)
+        let viewModel = ProductDownloadSettingsViewModel(product: model)
+        let defaultLimit: Int64 = -1
+
+        // Act
+        var isValidResult: Bool?
+        var shouldBringUpKeyboard: Bool?
+        waitForExpectation { exp in
+            viewModel.handleDownloadLimitChange("") { (isValid, shouldBringUpKeyboardValue) in
+                isValidResult = isValid
+                shouldBringUpKeyboard = shouldBringUpKeyboardValue
+                exp.fulfill()
+            }
+        }
```

Comment:
nit: I'd suggest using `let result = try waitFor { promise in ... }` which I find it simpler (no need to unwrap it!) - there are some examples in the codebase, Jayson first shared this great test helper to me :)